packages/discovery-react-components/src/components/StructuredQuery/components/AddRuleRowButton/AddRuleRowButton.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import { Button } from 'carbon-components-react';
+import Add16 from '@carbon/icons-react/lib/add/16';
+import { Messages } from 'components/StructuredQuery/messages';
+import { StructuredQuerySelection } from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+
+export interface AddRuleRowButtonProps {
+  /**
+   * text to display for the Add rule button
+   */
+  addRuleRowText: Messages['addRuleRowText'];
+  /**
+   * id of the group for the rule row to render, or 'top-level' if the top-level rule group
+   */
+  groupId: number;
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+export const AddRuleRowButton: FC<AddRuleRowButtonProps> = ({
+  addRuleRowText,
+  groupId,
+  structuredQuerySelection,
+  setStructuredQuerySelection
+}) => {
+  const handleOnClick = (
+    structuredQuerySelection: StructuredQuerySelection,
```

Comment:
i'm not sure you need to pass all these attributes into the function, you can just use them from the props above

Reply:
updated - this was related to moving this `handleOnClick` outside of the component when I split it out - thanks!